File: 242-ValidAnagram/242-ValidAnagram.java
// Last updated: 9/14/2025, 12:53:49 PM
class Solution {
    public boolean isAnagram(String s, String t) {
        int arrs[] = new int[26];
        if(s.length() != t.length()) return false;
        else{
            for(int i = 0; i < s.length(); i++){
                arrs[s.charAt(i)-'a']++;
            }
            for(int i = 0; i < t.length(); i++){
                arrs[t.charAt(i)-'a']--;
                if(arrs[t.charAt(i)-'a'] < 0) return false;
            }
            for(int i = 0; i < t.length(); i++){
                if(arrs[t.charAt(i)-'a'] != 0) return false;
            }
            return true;
        }
    }
}

File: 242-ValidAnagram/242-ValidAnagram.py
# Last updated: 9/14/2025, 12:53:46 PM
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        cnt=[0]*26
        cnt_t=[0]*26
        for ch in s:
            cnt[ord(ch)-ord('a')]+=1
        for ch in t:
            cnt_t[ord(ch)-ord('a')]+=1
        for i in range(0,26):
            if cnt[i]!=cnt_t[i]:return False
        return True